File: admin/src/main/java/hr/nas2skupa/eleventhhour/admin/CategoryViewHolder.java
package hr.nas2skupa.eleventhhour.admin;

import android.graphics.Color;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Callback;
import com.squareup.picasso.NetworkPolicy;
import com.squareup.picasso.Picasso;

import hr.nas2skupa.eleventhhour.model.Category;

/**
 * Created by nas2skupa on 18/09/16.
 */
public class CategoryViewHolder extends RecyclerView.ViewHolder {
    private TextView titleView;
    private ImageView iconView;
    private ImageView backgroundView;


    public CategoryViewHolder(View itemView) {
        super(itemView);

        titleView = (TextView) itemView.findViewById(R.id.txt_category_name);
        iconView = (ImageView) itemView.findViewById(R.id.img_category_icon);
        backgroundView = (ImageView) itemView.findViewById(R.id.category_background);
    }

    public void bindToCategory(final Category category) {
        titleView.setText(category.getLocalName());
        titleView.setSelected(true);
        try {
            backgroundView.setBackgroundColor(Color.parseColor(category.color));
        } catch (Exception ignored) {

        }
        Picasso.with(iconView.getContext()).load(category.icon)
                .networkPolicy(NetworkPolicy.OFFLINE)
                .fit()
                .into(iconView, new Callback() {
                    @Override
                    public void onSuccess() {
                    }

                    @Override
                    public void onError() {
                        //Try again online if cache failed
                        Picasso.with(iconView.getContext())
                                .load(category.icon)
                                .fit()
                                .into(iconView);
                    }
                });
    }
}

File: admin/src/main/java/hr/nas2skupa/eleventhhour/admin/viewholders/CategoryViewHolder.kt
package hr.nas2skupa.eleventhhour.admin.viewholders

import android.graphics.Color
import android.support.v7.widget.RecyclerView
import android.view.View
import com.squareup.picasso.Callback
import com.squareup.picasso.NetworkPolicy
import com.squareup.picasso.Picasso
import hr.nas2skupa.eleventhhour.common.model.Category
import kotlinx.android.synthetic.main.item_category.view.*

/**
 * Created by nas2skupa on 18/09/16.
 */
class CategoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    fun bindToCategory(category: Category) = with(itemView) {
        txt_category_name.text = category.localName
        txt_category_name.isSelected = true
        try {
            category_background.setBackgroundColor(Color.parseColor(category.color))
        } catch (ignored: Exception) {

        }

        Picasso.with(img_category_icon.context).load(category.icon)
                .networkPolicy(NetworkPolicy.OFFLINE)
                .fit()
                .into(img_category_icon, object : Callback {
                    override fun onSuccess() {}

                    override fun onError() {
                        //Try again online if cache failed
                        Picasso.with(img_category_icon.context)
                                .load(category.icon)
                                .fit()
                                .into(img_category_icon)
                    }
                })
    }
}
